src/main/kotlin/no/nav/syfo/personstatus/OversiktHendelseService.kt

```diff
@@ -8,136 +8,95 @@ import org.slf4j.LoggerFactory
-            }
-            else -> {
-                database.oppdaterPersonMedMoteplanleggerAlleSvarMottatt(oversiktHendelse)
-                COUNT_OVERSIKTHENDELSE_MOTEPLANLEGGER_ALLE_SVAR_MOTTATT_OPPDATER.increment()
-            }
-        }
-    }
-
-    private fun oppdaterPersonMedHendelseMoteplanleggerSvarBehandlet(oversiktHendelse: KOversikthendelse, callId: String) {
-        val person = database.hentPersonResultat(oversiktHendelse.fnr)
-        when {
-            person.isEmpty() -> {
-                log.error("Fant ikke person som skal oppdateres med hendelse {}, for enhet {}", oversiktHendelse.hendelseId, oversiktHendelse.enhetId)
-                COUNT_OVERSIKTHENDELSE_MOTEPLANLEGGER_ALLE_SVAR_BEHANDLET_FEILET.increment()
-            }
-            erPersonsEnhetOppdatert(person, oversiktHendelse.enhetId) -> {
-                database.oppdaterPersonMedMoteplanleggerAlleSvarBehandletNyEnhet(oversiktHendelse)
-                COUNT_OVERSIKTHENDELSE_MOTEPLANLEGGER_ALLE_SVAR_BEHANDLET_OPPDATER_ENHET.increment()
-            }
-            else -> {
-                database.oppdaterPersonMedMoteplanleggerAlleSvarBehandlet(oversiktHendelse)
-                COUNT_OVERSIKTHENDELSE_MOTEPLANLEGGER_ALLE_SVAR_BEHANDLET_OPPDATER.increment()
-            }
-        }
-    }
+        if (oversikthendelseType == null) {
+            log.error(
+                "Mottatt oversikthendelse med ukjent type, ${oversiktHendelse.hendelseId}, {}",
+                callIdArgument(callId)
+            )
+            COUNT_OVERSIKTHENDELSE_UKJENT_MOTTATT.increment()
+        } else {
+            val pPersonOversiktStatus = database.hentPersonResultat(
+                fnr = oversiktHendelse.fnr,
+            ).firstOrNull()
 
-    private fun oppdaterPersonMedHendelseMotebehovBehandlet(oversiktHendelse: KOversikthendelse, callId: String) {
-        val person = database.hentPersonResultat(oversiktHendelse.fnr)
-        when {
-            person.isEmpty() -> {
-                log.error("Fant ikke person som skal oppdateres med hendelse {}, for enhet {}, {}", oversiktHendelse.hendelseId, oversiktHendelse.enhetId, callIdArgument(callId))
-                COUNT_OVERSIKTHENDELSE_MOTEBEHOVSSVAR_BEHANDLET_FEILET.increment()
-            }
-            erPersonsEnhetOppdatert(person, oversiktHendelse.enhetId) -> {
-                database.oppdaterPersonMedMotebehovBehandletNyEnhet(oversiktHendelse)
-                COUNT_OVERSIKTHENDELSE_MOTEBEHOVSSVAR_BEHANDLET_OPPDATER_ENHET.increment()
-            }
-            else -> {
-                database.oppdaterPersonMedMotebehovBehandlet(oversiktHendelse)
-                COUNT_OVERSIKTHENDELSE_MOTEBEHOVSSVAR_BEHANDLET.increment()
+            if (pPersonOversiktStatus == null) {
+                if (oversikthendelseType.isNotBehandling()) {
+                    val newPersonoversiktStatusService = oversiktHendelse.toNewPersonOversikt(
```

Comment:
Kanskje man kan refaktorere ut create/update-koden her i egne metoder så blir det litt lettere å lese den nøstede ifen her.

Reply:
Enig, laget egen metode for create og update nå.

Reply:
Ser bra ut! Men kan droppe Service-suffix på `newPersonoversiktStatusService` eller?

Reply:
Ja, en typo der tror jeg.

Reply:
Fikset.